pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -212,22 +212,50 @@ func (r *ReconcileDeadmansSnitchIntegration) Reconcile(request reconcile.Request
 			return reconcile.Result{}, err
 		}
 
-		err = r.createSnitch(dmsi, &clusterdeployment, dmsc)
+		resourcesExist, err := r.snitchResourcesExist(dmsi, &clusterdeployment)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+		clusterIsHibernating := clusterdeployment.Spec.PowerState == hivev1.HibernatingClusterPowerState
 
-		err = r.createSecret(dmsi, dmsc, clusterdeployment)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
+		if clusterIsHibernating {
+			if resourcesExist {
+				err = r.deleteSnitch(dmsi, &clusterdeployment, dmsc)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
 
-		err = r.createSyncset(dmsi, clusterdeployment)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
+				dmsSecret := utils.SecretName(clusterdeployment.Spec.ClusterName, dmsi.Spec.SnitchNamePostFix)
+				err = utils.DeleteRefSecret(dmsSecret, clusterdeployment.Namespace, r.client)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+
+				err = utils.DeleteSyncSet(dmsSecret, clusterdeployment.Namespace, r.client)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
```

Comment:
Reuse existing delete logic.  `deleteDMSClusterDeployment(dmsi, clusterdeployment, dmsc)`

Reply:
I had seen that function, and it gets called if ClusterDeployment or DMSIntegration are deleted, updating the finalizers in the process.
I didn't think this situation should do that as we still want a snitch once we switch back out of hibernation. wdyt?

Reply:
If the cluster is hibernated it's not setup with DMS.  Which makes me think we could approach this a different way.  If we had a label on the CD we could update the DMSI selector to exclude clusters that are hibernated.  Then no special logic in the operator.  I don't think there's a valid reason to keep the finalizer around when there isn't any DMS service.  If someone issued a delete on the CD when hibernated, DMS shouldn't be involved.

Reply:
@jewzaam Ok, cheers. As for this work, I have made that final change to use `deleteDMSClusterDeployment(dmsi, clusterdeployment, dmsc)` and tidied up the commit history.

As for the selector, it's a fair suggestion. I did check a hibernating ClusterDeployment CR in stage and there is currently no label applied - so that would be work needed in OCM side.
Pros for this approach is if we need to refine the hibernation in the future, the code is here and just needs updating. This would be a decision for the dms operator team to make. wdyt?

Reply:
@jewzaam (reping on above :smile: )